src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.*;
+
+/**
+ * Represents a class that takes in inputs and translates them into different commands.
+ */
+public class Parser {
+
+    /**
+     * Takes in a string input and convert the input into a command which performs a set of
+     * instructions on the task list, ui and storage.
+     *
+     * @param input String that contains the input of the user.
+     * @return A command that execute a set of instructions.
+     * @throws DukeException Thrown when there is a Duke exception.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] arguments = input.split(" ");
+
+        switch (arguments[0]) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "done":
+        case "delete":
+            int num;
+            try {
+                num = Integer.parseInt(arguments[1]);
+            } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+                throw new IntFormatException(arguments[0]);
+            }
+            if (arguments[0].equals("done")) {
+                return new DoneCommand(num);
+            } else {
+                return new DeleteCommand(num);
+            }
+        case "deadline":
+        case "event":
+        case "todo":
```

Comment:
Sorry if I sound incompetent, but I didn't know you can do this! 👍 I have been thinking how to not duplicate codes for this type of segments.

Reply:
Glad that you have learnt something from my code! 😄 This works because the switch allows fall through. Apparently there is also a thing called labels in java which allows you to do things like "goto" in order languages. You can read more about it on this link: https://www.tutorialspoint.com/How-to-use-labels-in-Java-code